internal/throttler/remote/throttler.go

```diff
@@ -102,10 +100,13 @@ func NewThrottler(service string, options ...Option) *Throttler {
 func (t *Throttler) IsAllowed(operation string) bool {
 	t.mux.Lock()
 	defer t.mux.Unlock()
-	_, ok := t.credits[operation]
-	if !ok {
-		if !t.synchronousInitialization {
+	value, ok := t.credits[operation]
+	if !ok || value == 0 {
+		if !ok {
```

Comment:
@black-adder you might notice this changes the throttler to set operations to zero whenever they are encountered, ignoring the `t.synchronousInitialization` flag. The reason for this is to maintain the operations for the request that requires a list of operation names. Furthermore, that is why I changed the condition above to check for `!ok || value == 0`. When `value` is `0`, it is either not initialized and just a placeholder, or it needs more credits. Either way, that indicates it is a good time to consider a synchronous request.